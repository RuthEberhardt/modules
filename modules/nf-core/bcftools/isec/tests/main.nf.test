nextflow_process {

    name "Test Process BCFTOOLS_ISEC"
    script "../main.nf"
    process "BCFTOOLS_ISEC"

    tag "modules"
    tag "modules_nfcore"
    tag "bcftools"
    tag "bcftools/isec"

    config "./nextflow.config"

    test("sarscov2 - [[vcf1.gz, vcf2.gz], [tbi1, tbi2]]") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test.vcf.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test2.vcf.gz', checkIfExists: true)
                        ],
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test.vcf.gz.tbi', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test2.vcf.gz.tbi', checkIfExists: true)
                        ]
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - [[vcf1, vcf2], [tbi1, tbi2]] - test") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test.vcf.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test2.vcf.gz', checkIfExists: true)
                        ],
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test.vcf.gz.tbi', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test2.vcf.gz.tbi', checkIfExists: true)
                        ]
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
