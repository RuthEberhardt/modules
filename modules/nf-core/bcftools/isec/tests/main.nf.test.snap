{
    "sarscov2 - [[vcf1.gz, vcf2.gz], [tbi1, tbi2]]": {
        "content": [
            [
                "README.txt",
                "0000.vcf.gz",
                "0001.vcf.gz",
                "sites.txt",
                "0000.vcf.gz.tbi",
                "0001.vcf.gz.tbi"
            ],
            [
                "0000.vcf.gz:md5,8e722884ffb75155212a3fc053918766",
                "0001.vcf.gz:md5,b39a72f91458b94b346dd73690207649",
                "README.txt:md5,10fc33b66522645600d44afbd41fb792",
                "sites.txt:md5,1cea3fbde7f6d3c97f3d39036f9690df"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-03T15:30:13.99495369"
    },
    "sarscov2 - [[vcf1, vcf2], [tbi1, tbi2]] - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "0000.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "0000.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "0001.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "0001.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "README.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "sites.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,c5b5a1204cf56fec79d17e1913eaca27"
                ],
                "results": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "0000.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "0000.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "0001.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "0001.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "README.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "sites.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c5b5a1204cf56fec79d17e1913eaca27"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-03T09:45:52.881120953"
    }
}